packages/jsii/lib/assembler.ts

```diff
@@ -892,6 +894,45 @@ export class Assembler implements Emitter {
     return [jsiiType];
   }
 
+  private _validateHeritageClauses(clauses?: ts.NodeArray<ts.HeritageClause>) {
+    if (clauses == null || clauses.length === 0) {
+      // Nothing to do.
+      return;
+    }
+    for (const clause of clauses) {
+      for (const node of clause.types) {
+        const parentType = this._typeChecker.getTypeAtLocation(node);
+        // For some reason, we cannot trust parentType.isClassOrInterface()
+        const badDecl = parentType.symbol.declarations.find(
+          (decl) =>
+            !ts.isClassDeclaration(decl) && // <-- local classes
+            !ts.isInterfaceDeclaration(decl) && // <-- local interfaces
+            !ts.isModuleDeclaration(decl), // <-- imported types
+        );
+        if (badDecl != null) {
+          this._diagnostic(
+            node,
+            ts.DiagnosticCategory.Error,
+            `Illegal "${clauseType(clause.token)}" value for an exported API: ${
+              ts.SyntaxKind[badDecl.kind]
+            }`,
```

Comment:
What does this error look like the `neg.omit.*.ts`? 

Make sure it's easy to understand for an external contributor to the CDK.

Reply:
```
Illegal "extends" value for an exported API: Omit<SomeInterface, 'somefield'>
```

I'd have wanted to make it more specific (mapped/utility types are not allowed here), however the type checker does not really give me a way to do this with confidence...

Reply:
This looks fairly good.